dbapi/src/main/java/com/griddynamics/jagger/dbapi/provider/SessionInfoProviderImpl.java

```diff
@@ -273,53 +283,61 @@ public Long getFirstPosition(Set<String> selectedIds) throws RuntimeException {
         checkArgument(length >= 0, "length is negative");
         checkNotNull(sessionIds, "sessionIds is null");
 
-        long timestamp = System.currentTimeMillis();
+        boolean isFetchAll = sessionIds.size() == 0; // if input set is empty - fetch all session data.
 
+        long timestamp = System.currentTimeMillis();
         List<SessionDataDto> sessionDataDtoList;
-
         try {
+            Query query = entityManager.createQuery(
+                    "select sd from SessionData as sd " +
+                    (isFetchAll ? " " : "where sd.sessionId in (:sessionIds) ") +
+                    "order by sd.startTime asc"
+            ).setFirstResult(start);
+            if (!isFetchAll) {
+                query.setParameter("sessionIds", new ArrayList<String>(sessionIds))
+                     .setMaxResults(length);
+            }
             @SuppressWarnings("unchecked")
-            List<SessionData> sessionDataList = (List<SessionData>)
-                    entityManager.createQuery("select sd from SessionData as sd where sd.sessionId in (:sessionIds) order by sd.startTime asc")
-                            .setParameter("sessionIds", new ArrayList<String>(sessionIds))
-                            .setFirstResult(start)
-                            .setMaxResults(length)
-                            .getResultList();
-
+            List<SessionData> sessionDataList = (List<SessionData>) query.getResultList();
             if (sessionDataList.isEmpty()) {
                 return Collections.<SessionDataDto>emptyList();
             }
 
             Map<Long, String> userCommentMap = Collections.EMPTY_MAP;
-
             if (isUserCommentStorageAvailable) {
-
-                List<Object[]> userComments = entityManager.createQuery(
-                        "select smd.sessionData.id, smd.userComment from SessionMetaDataEntity as smd where smd.sessionData in (:sessionDataList)")
-                        .setParameter("sessionDataList", sessionDataList)
-                        .getResultList();
-
+                query = entityManager.createQuery(
+                        "select smd.sessionData.id, smd.userComment from SessionMetaDataEntity as smd "
+                        + (isFetchAll ? " " : "where smd.sessionData in (:sessionDataList)")
+                );
+                if (!isFetchAll) {
+                    query.setParameter("sessionDataList", sessionDataList);
+                }
+                List<Object[]> userComments = query.getResultList();
                 if (!userComments.isEmpty()) {
```

Comment:
1. You left a comment on a piece of code not related to my commit but just visible in this change.
2. It is an arguable statement. I'd like to leave as is because it is more concise and is more like english phrase -> more readable to me